code/datums/action.dm

```diff
@@ -502,15 +502,15 @@
 	button.moved = "6:157,4:-2"
 
 /datum/action/item_action/cult_dagger/Trigger()
-	for(var/obj/item/melee/cultblade/dagger/held_item in owner.held_items) // In case we were already holding a dagger
-		held_item.attack_self(owner)
-		return
-	var/obj/item/target_item = target
-	if(owner.can_equip(target_item, ITEM_SLOT_HANDS))
-		owner.temporarilyRemoveItemFromInventory(target_item)
-		owner.put_in_hands(target_item)
-		target_item.attack_self(owner)
-		return
+	for(var/obj/item/H in owner.held_items) //In case we were already holding another dagger
+		if(istype(H, /obj/item/melee/cultblade/dagger))
```

Comment:
I genuinely cannot understand what this is meant to to, it's literally making the proc worse, since the type filtering is just slower

Reply:
Hmm tested it, doesn't seem to change anything? I'll revert it then